### PR TITLE
macvlan networks

### PR DESCRIPTION
**Pull request by baude**
add the ability to a macvlan network with podman network create.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: docs/source/markdown/podman-network-create.1.md [hunk 1]
```diff
@@ -29,6 +31,11 @@ Driver to manage the network (default "bridge").  Currently on `bridge` is suppo
 Define a gateway for the subnet. If you want to provide a gateway address, you must also provide a
 *subnet* option.
 
+**--host-device**
+
+Host networking device that will be used by *macvlan* connections.  This option must be used with the
```
Comment (rhatdan) [inline docs/source/markdown/podman-network-create.1.md]: Should this be plural, since you can add one or more devices for a vlan?

Comment (baude) [reply docs/source/markdown/podman-network-create.1.md]: you can only add one.  shall i singularize the language in the man page. is that what confused this?

Comment (rhatdan) [reply docs/source/markdown/podman-network-create.1.md]: The code allows users to add multiple. Should we combine the options, into one.  
--macvlanaddr="1.2.3.4"
Which would indicate that you want to use macvlan and don't have to specify the address separately. removes extra error checking.

File: docs/source/markdown/podman-network-create.1.md [hunk 2]
```diff
@@ -38,6 +43,10 @@ Restrict external access of this network
 Allocate container IP from a range.  The range must be a complete subnet and in CIDR notation.  The *ip-range* option
 must be used with a *subnet* option.
 
+**--macvlan**
+
+Create a *macvlan* based connection rather than a classic bridge based on the given host-device.
```
Comment (rhatdan) [inline docs/source/markdown/podman-network-create.1.md]: You should add an example of using this.

**Comment by rhatdan** [comment]
Looks good other then docs.  It might be just me, but I would need more explanation of what is going on in the man page.

File: docs/source/markdown/podman-network-create.1.md [hunk 3]
```diff
@@ -29,6 +31,9 @@ Driver to manage the network (default "bridge").  Currently on `bridge` is suppo
 Define a gateway for the subnet. If you want to provide a gateway address, you must also provide a
 *subnet* option.
 
+**--host-device**
```
Comment (mheon) [inline docs/source/markdown/podman-network-create.1.md]: Description?

File: docs/source/markdown/podman-network-create.1.md [hunk 4]
```diff
@@ -7,8 +7,10 @@ podman\-network-create - Create a Podman CNI network
 **podman network create**  [*options*] name
 
 ## DESCRIPTION
-Create a CNI-network configuration for use with Podman. At the time of this writing, the only network
-type that can be created is a *bridge* network.
+Create a CNI-network configuration for use with Podman. By default, Podman creates a bridge connection. A
+*macvlan* connection can be created with the *macvlan* option. In the case of *macvlan* connections, the
+CNI *dhcp* plugin needs to be activated or the container image must have a DHCP client to interact
```
Comment (mheon) [inline docs/source/markdown/podman-network-create.1.md]: How do we activate the DHCP plugin?

File: docs/source/markdown/podman-network-create.1.md (hunk 2 above)
Comment (mheon) [inline docs/source/markdown/podman-network-create.1.md]: This is a little unclear - "based on the given host device" could apply to "classic bridge" or "macvlan based connection"

File: cmd/podman/network_create.go [hunk 5]
```diff
@@ -41,6 +41,7 @@ func init() {
 	flags.IPVar(&networkCreateCommand.Gateway, "gateway", nil, "IPv4 or IPv6 gateway for the subnet")
 	flags.BoolVar(&networkCreateCommand.Internal, "internal", false, "restrict external access from this network")
 	flags.IPNetVar(&networkCreateCommand.IPRange, "ip-range", net.IPNet{}, "allocate container IP from range")
+	flags.StringVar(&networkCreateCommand.MacVLAN, "macvlan", "", "create a macvlan connection based on this device")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/network_create.go]: The majority of references, including Docker refs,  that I see for macvlan have an upper case 'M", ala "Macvlan.  We might want to consider that usage.

Comment (baude) [reply cmd/podman/network_create.go]: options for us begin with lower case

File: docs/source/markdown/podman-network-create.1.md (hunk 3 above)
Comment (baude) [reply docs/source/markdown/podman-network-create.1.md]: this should be removed, err on my part

File: pkg/adapter/network.go [hunk 6]
```diff
@@ -262,3 +262,50 @@ func (r *LocalRuntime) NetworkCreate(cli *cliconfig.NetworkCreateValues) (string
 	err = ioutil.WriteFile(cniPathName, b, 0644)
 	return cniPathName, err
 }
+
+// NetworkCreateBridge creates a CNI network
```
Comment (haircommander) [inline pkg/adapter/network.go]: copy paste error

File: cmd/podman/network_create.go (hunk 5 above)
Comment (mheon) [reply cmd/podman/network_create.go]: He's talking about the description

File: docs/source/markdown/podman-network-create.1.md (hunk 4 above)
Comment (baude) [reply docs/source/markdown/podman-network-create.1.md]: that is out of scope for podman ... that said, i think our fedora cni package should ship the systemd service file

**Comment by baude** [comment]
fixed up things but as I said at the f2f meeting, adding macvlan does not fit the structure of this command very well.  It makes the whole thing little unbalanced.

File: cmd/podman/network_create.go (hunk 5 above)
Comment (TomSweeneyRedHat) [reply cmd/podman/network_create.go]: As @mheon pointed out, the option is fine, it's the description.

File: docs/source/markdown/podman-network-create.1.md [hunk 7]
```diff
@@ -38,6 +40,11 @@ Restrict external access of this network
 Allocate container IP from a range.  The range must be a complete subnet and in CIDR notation.  The *ip-range* option
 must be used with a *subnet* option.
 
+**--macvlan**
+
+Create a *macvlan* based connection rather than a classic bridge.  You must pass an interface name from the host for the
+macvlan connection.
+
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-network-create.1.md]: Other than the option itself, I'd change all the "macvlan" refs to "Macvlan".

**Pull request merged**